pkg/discovery/worker/dispatcher.go

```diff
@@ -203,18 +207,18 @@ func (d *SamplingDispatcher) dispatchSamplingDiscoveries(nodes []*api.Node, samp
 			glog.V(3).Infof("Dispatching sampling discovery task %v", currTask)
 			d.assignTask(currTask)
```

Comment:
Ideally, we should abort as well while waiting for the next available worker.  So it seems to me the select/abort logic should be inside the `assignTask()` func.

Reply:
Agreed, that makes sense.  Thanks @ading1977!